go/host/enclave/guardian.go

```diff
@@ -470,14 +470,24 @@ func (g *Guardian) processL1BlockTransactions(block *common.L1Block) {
 		if err != nil {
 			g.logger.Error("Could not decode rollup.", log.ErrKey, err)
 		}
-		err = g.db.AddRollupHeader(r, block)
+
+		metaData, err := g.enclaveClient.GetRollupData(r.Header.Hash())
+		if err != nil {
+			g.logger.Error("Could not fetch rollup metadata from enclave.", log.ErrKey, err)
```

Comment:
If there was an error here I'm guessing you don't want to add it to the db on the next line, because metadata will be nil.

Reply:
Good spot!